lib/models/record.js

```diff
@@ -77,6 +78,33 @@ class Record extends BaseModel {
       (r) => new Server({ id: r.record_info.server_id })
     );
   }
+
+  async splits() {
+    if (this.rank > 1) return [];
+
+    let splits =
+      this.attrs.splits || (await this.access("splits", (r) => r.splits));
+
+    if (!splits) {
+      let rs = null;
+      if (this.attrs.map.attrs.name) {
```

Comment:
I think this is basically right, although for now I think it can also come from `this.attrs.map_info.id` or `this.attrs.map_info.name`, so if you just `||` with those in the corresponding spots I think it should work?

Reply:
Looks like `this.attrs.class` can also be either `3/4` or `"soldier"/"demoman"` depening on where this is called from.